source/loader/ur_adapter_registry.hpp

```diff
@@ -152,10 +152,101 @@ class AdapterRegistry {
         return paths.empty() ? std::nullopt : std::optional(paths);
     }
 
+    ur_result_t readPreFilterODS(std::string platformBackendName) {
+        bool acceptLibrary = true;
+        auto odsEnvMap = getenv_to_map("ONEAPI_DEVICE_SELECTOR", false);
+        logger::debug(
+            "getenv_to_map parsed env var and {} a map",
+            (odsEnvMap.has_value() ? "produced" : "failed to produce"));
+
+        // if the ODS env var is not set at all, then pretend it was set to the default
+        using EnvVarMap = std::map<std::string, std::vector<std::string>>;
```

Comment:
Agreed, I had to change only a minor portion of this related to only filtering the backend types and using a substring lookup, but otherwise it is much of the same code. I will add a TODO to move this to the common as a refactor.

Reply:
Add the todo in the latest change.